src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/DynamicDebuggerProxy.cs

```diff
@@ -0,0 +1,482 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Dynamic;
+using System.Linq.Expressions;
+using System.Runtime.CompilerServices;
+using System.Runtime.Serialization;
+
+namespace Microsoft.CSharp.RuntimeBinder
+{
+    [Serializable]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    internal sealed class DynamicBindingFailedException : Exception
+    {
+        public DynamicBindingFailedException()
+            : base()
+        {
+        }
+
+        private DynamicBindingFailedException(SerializationInfo info, StreamingContext context) 
+            : base(info, context) 
+        { 
+        }
+    }
+
+    internal sealed class GetMemberValueBinder : GetMemberBinder
+    {
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Performance", "CA1811:AvoidUncalledPrivateCode")]
+        public GetMemberValueBinder(string name, bool ignoreCase)
+            : base(name, ignoreCase)
+        {
+        }
+
+        public override DynamicMetaObject FallbackGetMember(DynamicMetaObject self, DynamicMetaObject onBindingError)
+        {
+            if (onBindingError == null)
+            {
+                var v = new List<DynamicMetaObject> { self };
+                var error = new DynamicMetaObject(System.Linq.Expressions.Expression.Throw(
+                    System.Linq.Expressions.Expression.Constant(new DynamicBindingFailedException(), typeof(Exception)), typeof(object)), System.Dynamic.BindingRestrictions.Combine(v));
+                return error;
+            }
+            return onBindingError;
+        }
+    }
+
+    internal sealed class DynamicMetaObjectProviderDebugView
+    {
+        [System.Diagnostics.DebuggerDisplay("{value}", Name = "{name, nq}", Type = "{type, nq}")]
+        internal class DynamicProperty
+        {
+            [System.Diagnostics.DebuggerBrowsable(System.Diagnostics.DebuggerBrowsableState.Never)]
+            [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Performance", "CA1823:AvoidUnusedPrivateFields")]
+            readonly string name;
+
+            [System.Diagnostics.DebuggerBrowsable(System.Diagnostics.DebuggerBrowsableState.Never)]
+            [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Performance", "CA1823:AvoidUnusedPrivateFields")]
+            readonly object value;
+
+            [System.Diagnostics.DebuggerBrowsable(System.Diagnostics.DebuggerBrowsableState.Never)]
+            [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Performance", "CA1823:AvoidUnusedPrivateFields")]
+            readonly string type;
```

Comment:
Nit: could we add in the visibility explicitly (`private`)?

Also, I assume we don't want to change these names because their whole purpose is to be displayed in the debugger and we want them to be e.g. "name" instead of "_name"?